scvid/data/distributed_anndata.py

```diff
@@ -36,6 +38,38 @@ def lazy_getattr():
         _GETATTR_MODE.lazy = False
 
 
+class DistributedAnnCollectionView(AnnCollectionView):
+    def __getitem__(self, index: Index) -> DistributedAnnCollectionView:
+        oidx, vidx = _normalize_indices(index, self.obs_names, self.var_names)
+        resolved_idx = self._resolve_idx(oidx, vidx)
+
+        return DistributedAnnCollectionView(self.reference, self.convert, resolved_idx)
+
+    @property
+    def obs_names(self) -> pd.Index:
+        """
+        Gather and return the obs_names from all AnnData objects in the collection.
+        """
+        indices = []
+        for i, oidx in enumerate(self.adatas_oidx):
+            if oidx is None:
+                continue
+
+            adata = self.adatas[i]
+            indices.append(adata.obs_names[oidx])
+
+        if len(indices) > 1:
+            concat_indices = pd.concat(
+                [pd.Series(idx) for idx in indices], ignore_index=True
+            )
+            obs_names = pd.Index(concat_indices)
+            obs_names = obs_names if self.reverse is None else obs_names[self.reverse]
```

Comment:
Reverse order if necessary. This ensures that `.X` and `.obs_names` have the same order.